### PR TITLE
Week5/recommendation submission

### PR DESCRIPTION
**Pull request by claudiachua**
1. Add Recommendation Container, it will reload when user change the groupID on dropdownlist. 
2. Add a recommendation form with 2 hidden input field: userEmail & groupId
3. Recommendation is stored in datastore with the entity key of userEmail:groupId (E.g. test@example.com:123)

### CODE REVIEW

**Review by riku1212 — APPROVED** [review]
LGTM +1

File: friendly/src/main/webapp/script.js [hunk 1]
```diff
@@ -47,8 +49,11 @@ function getUserGroups() {
         else {
             groupData.innerHTML = ('<h4>Your groups: </h4>');
             groupDropdown = document.createElement('select');
+            groupDropdown.setAttribute("id", "groupId");
+            groupDropdown.setAttribute("onchange","getRecommendationContainer()"); //this is not working
```
Comment (riku1212) [inline friendly/src/main/webapp/script.js]: i think `groupDropdown.onchange = getRecommendationContainer()` should work

**Review by riku1212 — CHANGES_REQUESTED** [review]

File: friendly/src/main/webapp/script.js (hunk 1 above)
Comment (claudiachua) [reply friendly/src/main/webapp/script.js]: Oops. groupDropdown.setAttribute() works, I forget to remove the comment

**Review by riku1212 — APPROVED** [review]

File: friendly/src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        String location = request.getParameter("location");
+        String userEmail = request.getParameter("userEmail");
+
+        ArrayList<String> recommendation = new ArrayList<String>();
+        recommendation.add(restaurantName);
+        recommendation.add(location);
+
+        //Add new restaurant to datastore
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity groupEntity = new Entity(userEmail + ":" + groupId);
```
Comment (riku1212) [inline friendly/src/main/java/com/google/sps/servlets/RecommendationServlet.java:32]: just wondering, why use this kind of entity to store the data? don't you think it's better to use level of parents?
https://cloud.google.com/datastore/docs/concepts/entities

**Pull request merged**